TWLight/tests.py

```diff
@@ -429,22 +408,436 @@ def test_email_required_2(self):
+        self.partner1.save()
+        self.partner2.coordinator = self.editor4.user
+        self.partner2.save()
+        self.partner3.coordinator = self.editor4.user
+        self.partner3.save()
+        self.partner4.coordinator = self.editor4.user
+        self.partner4.save()
+
+        # Editor 5 is a coordinator without a session and with no designated partners.
+        self.editor5 = EditorFactory()
+        coordinators.user_set.add(self.editor5.user)
+
+        # Create applications.
+        self.app1 = ApplicationFactory(
+            editor=self.editor1, partner=self.partner1, status=Application.PENDING
+        )
+        self.app2 = ApplicationFactory(
+            editor=self.editor2, partner=self.partner1, status=Application.PENDING
+        )
+        self.app3 = ApplicationFactory(
+            editor=self.editor3, partner=self.partner1, status=Application.PENDING
+        )
+        self.app4 = ApplicationFactory(
+            editor=self.editor1, partner=self.partner2, status=Application.PENDING
+        )
+        self.app5 = ApplicationFactory(
+            editor=self.editor2, partner=self.partner2, status=Application.PENDING
+        )
+        self.app6 = ApplicationFactory(
+            editor=self.editor3, partner=self.partner2, status=Application.PENDING
+        )
+        self.app7 = ApplicationFactory(
+            editor=self.editor1, partner=self.partner3, status=Application.PENDING
+        )
+        self.app8 = ApplicationFactory(
+            editor=self.editor1, partner=self.partner4, status=Application.PENDING
+        )
+        self.app9 = ApplicationFactory(
+            editor=self.editor2, partner=self.partner3, status=Application.PENDING
+        )
+
+        # Editor 4 will update status on applications to partners 1 and 2.
+        # Send the application
+        self.client.post(
+            reverse("applications:evaluate", kwargs={"pk": self.app1.pk}),
+            data={"status": Application.SENT},
+            follow=True,
+        )
+        self.app1.refresh_from_db()
+        self.auth_app1 = Authorization.objects.get(
+            authorizer=self.editor4.user,
+            authorized_user=self.editor1.user,
+            partner=self.partner1,
+        )
+
+        # Approve the application
+        self.client.post(
+            reverse("applications:evaluate", kwargs={"pk": self.app2.pk}),
+            data={"status": Application.APPROVED},
+            follow=True,
+        )
+        self.app2.refresh_from_db()
+        self.auth_app2 = Authorization(
+            authorizer=self.editor4.user,
+            authorized_user=self.editor2.user,
+            partner=self.partner1,
+        )
+
+        # Send the application
+        self.client.post(
+            reverse("applications:evaluate", kwargs={"pk": self.app3.pk}),
+            data={"status": Application.SENT},
+            follow=True,
+        )
+        self.app3.refresh_from_db()
```

Comment:
Is it reliable to do client POSTs in setUp? Feels like we should make the objects directly for reliability, but I'm not sure.

Reply:
should be totes reliable. I like doing the process in the same number of steps as the real world, e.g. create the pending application, then have an coordinator approve it. Doing it that way creates all of the appropriate revisions of the application. I could switch the over to a factory on the evaluate view instead of calling the client to make it more like the tests that follow. I had a vague recollection that factory requests had some limitations that came into play with the application revisions, but I'll go double-check.

Reply:
Just went through and tested doing factory request posts in setup, and those don't work. The client-based app updates work just fine in setup though.

Reply:
dug into it just a little more to see why, and basically the answer is authentication. I could go through and patch up the factory coordinator and the factory requests, but that's basically exactly what the client is for.